data/transactions/logic/eval.go

```diff
@@ -4060,3 +4061,109 @@ func opBase64Decode(cx *EvalContext) {
 	}
 	cx.stack[last].Bytes, cx.err = base64Decode(cx.stack[last].Bytes, encoding)
 }
+func hasDuplicateKeys(jsonText []byte) (bool, error) {
+	dec := json.NewDecoder(bytes.NewReader(jsonText))
+	keys := make(map[string]bool)
+	var value interface{}
+	_, err := dec.Token()
+	if err != nil {
+		return false, err
+	}
+	for dec.More() {
+		// Get JSON key
+		key, err := dec.Token()
+		if err != nil {
+			return false, err
+		}
+		// end of json
+		if key == '}' {
```

Comment:
does it mean no nested objects?

Reply:
right. it's only checking duplicate keys at top level. json_ref only retrieve value one level at a time.